crater: assert_eq!(NLL, location-insensitive polonius);

lqd opened the pull request:
Crater run to check `-Zpolonius=next`. Any differences in how NLLs and location-insensitive polonius compute scopes will cause an assert to trip and an ICE.

r? @ghost

lqd committed:
tmp: turn `-Zpolonius=next` on by default

lqd commented:
@bors try

bors commented:
:hourglass: Trying commit 16af7f5bdf1eced2cbeb3f1419950aec68e5341b with merge 031b389f0c31c0f91c9464d26313ae513dbe8d36...
<!-- homu: {"type":"TryBuildStarted","head_sha":"16af7f5bdf1eced2cbeb3f1419950aec68e5341b","merge_sha":"031b389f0c31c0f91c9464d26313ae513dbe8d36"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/6757477629/job/18368742895)
Build commit: 031b389f0c31c0f91c9464d26313ae513dbe8d36 (`031b389f0c31c0f91c9464d26313ae513dbe8d36`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/6757477629/job/18368742895"},"merge_sha":"031b389f0c31c0f91c9464d26313ae513dbe8d36"} -->

lqd commented:
@craterbot check

lqd commented:
WIP triage until I can properly dedupe the root causes:
- "No space left on device": 48
- most of the rest are differences in loan scopes

Some hacky and quick deduplication on crates with loan differences:

 - [`ai-cup-22`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/gh/SHaaD94.AICup2022/log.txt)
 - [`dev_bestia_simple_server`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/gh/LucianoBestia.cargo_crev_reviews/log.txt)
 - [`git2`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/reg/git2-0.18.1/log.txt)
 - [`nekoton`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/gh/broxus.nekoton-wasm/log.txt)
 - [`ppjson`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/gh/j0no.ppjson/log.txt)
 - [`simple-server`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/gh/AnyTimeTraveler.pipes-and-rust/log.txt)
 - [`simple-test-bbarekas`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/reg/simple-test-bbarekas-0.1.3/log.txt)
 - [`sp-state-machine`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/gh/Allfeat.genres-registry/log.txt)
 - [`spirt`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/reg/spirt-0.3.0/log.txt)
 - [`wavefc-cli`](https://crater-reports.s3.amazonaws.com/pr-117593/try%23031b389f0c31c0f91c9464d26313ae513dbe8d36/reg/wavefc-cli-3.0.0/log.txt)

lqd committed:
tmp: move and maintain live loans in `LivenessValues`

Liveness data is pushed from multiple parts of NLL. Instead of changing
the call sites to maintain live loans, move the latter to `LivenessValues` where
this liveness data is pushed to, and maintain live loans there.

This fixes the differences in polonius scopes on some CFGs where a
variable was dead in tracing but as a MIR terminator its regions were marked
live from "constraint generation"

bors commented:
:hourglass: Trying commit 952ab1995ab00435dcdebd2a7c24a6bd56ad124b with merge 08b2c37c39cf0d12f78ab70b2b8811ea41390c90...
<!-- homu: {"type":"TryBuildStarted","head_sha":"952ab1995ab00435dcdebd2a7c24a6bd56ad124b","merge_sha":"08b2c37c39cf0d12f78ab70b2b8811ea41390c90"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/6956894781/job/18931612978)
Build commit: 08b2c37c39cf0d12f78ab70b2b8811ea41390c90 (`08b2c37c39cf0d12f78ab70b2b8811ea41390c90`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/6956894781/job/18931612978"},"merge_sha":"08b2c37c39cf0d12f78ab70b2b8811ea41390c90"} -->

lqd commented:
Short run, and will also filter out the disk space related issues.

@craterbot check p=1 crates=https://crater-reports.s3.amazonaws.com/pr-117593/retry-regressed-list.txt

lqd commented:
sweet
